conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -154,4 +139,33 @@ object DynamicInterpreter {
     }
   }
 
+  private def getStepsAndExplode(conformance: ConfDataset, inputDf: Dataset[Row], experimentalMappingRule: Boolean):
```

Comment:
Is this method extraction necessary?
1. Method names with `And` in them make me think a method is doing too much.
2. The expanded nesting of the `experimentalMappingRule` explosions makes me a little worried that we might be introducing more complexity than we are reducing by refactoring right now.